mongoo/mongoo.go

```diff
@@ -0,0 +1,106 @@
+package mongoo
```

Comment:
I believe that creating a new package can bring more confusion than before, I think it would be better to keep this implementation together with what we have today, in the end both `mgo` and `mongo-driver` live in the same package.

What do you think?

Reply:
Do you mean to say that we need to replace mgo with mongo-driver? I agree with you. I created this(mongoo) as a temporary package with the plan to discuss this with you. If you're ok to replace mgo with mongo-driver, I will make the changes.

Reply:
I don't think that we should replace the `mgo`, instead we need to find a way to work with both.

Reply:
I saw that `mgo` has become unmaintained, it's a good opportunity to replace it. In that case I think we could move this code to `mongo` package and rename the files to `mongodb`.

With that we will have both drivers, and to differentiate one from another, the `New` method here, at least for now, should be replaced by `NewMongoDriver`.

Putting both together will gives an opportunity to add the `deprecation` flag in the old one.

Reply:
Yes, mgo seems be unmaintained. Good to have the both `mgo` and `mongo` and add deprecated flag to `mgo`. I have done the following changes. I removed `mongoo` package and moved all the files to the same `mongo` package. Renamed files to match the drivers name(`mgo` and `mongo`). 

1. Renamed mgo files and struct
2. Renamed mongo files and struct
3. Added NewMongoDriver for official mongodb driver
4. Added deprecated flag to mongo.New()

Please check the code.

Reply:
Great, I think you just forgot to rename the `mongoo` to `mongo`, I promise that will be the final changes 😄 .

Reply:
Sorry 😄, I forgot to mention it here but I did it. I renamed `mongoo.go` and `mongoo_test.go` to `mongo.go` and `mongo_test.go`.  Now, both implementation (old and new) resides in the same package `mongo`.  I changed the old files (mongo.go and mongo_test.go) to `mgo.go` and `mgo_test.go`

Reply:
Ah sorry 😅 , I meant the struct name, in the new `mongo.go` has the struct with the old `mongoo`. Also after you finish the changes please make a squash of the commits.

Thanks for the contribution.

Reply:
Hi @faabiosr 
I've done the changes. Please check and let me know if it needs any changes 😄
Thanks

Reply:
Hi Mate! Sorry for the delay, I was dealing with some personal stuff these days. 

Everything you've done here is amazing. Thanks for the contribution!